samples/cpp/text2image/lora_fuse.cpp

```diff
@@ -0,0 +1,48 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "openvino/genai/image_generation/text2image_pipeline.hpp"
+
+#include "imwrite.hpp"
+
+int32_t main(int32_t argc, char* argv[]) try {
+    OPENVINO_ASSERT(argc >= 3 && (argc - 3) % 2 == 0, "Usage: ", argv[0], " <MODEL_DIR> '<PROMPT>' [<LORA_SAFETENSORS> <ALPHA> ...]]");
+
+    const std::string models_path = argv[1], prompt = argv[2];
+    const std::string device = "CPU";  // GPU can be used as well
+
+    // MODE_FUSE instructs pipeline to fuse adapter tensors into original model weights loaded into memory
+    // giving the same performance level for inference as for the original model. After doing it you cannot
+    // change adapter dynamically without re-initializing the pipeline from scratch.
+    ov::genai::AdapterConfig adapter_config(ov::genai::AdapterConfig::MODE_FUSE);
+
+    // Multiple LoRA adapters applied simultaneously are supported, parse them all and corresponding alphas from cmd parameters:
+    for(size_t i = 0; i < (argc - 3)/2; ++i) {
+        ov::genai::Adapter adapter(argv[3 + 2*i]);
+        float alpha = std::atof(argv[3 + 2*i + 1]);
+        adapter_config.add(adapter, alpha);
+    }
+
+    // LoRA adapters passed to the constructor will be activated by default in next generates
+    ov::genai::Text2ImagePipeline pipe(models_path, device, ov::genai::adapters(adapter_config));
+
+    std::cout << "Generating image with LoRA adapters fused into original weights, resulting image will be in lora.bmp\n";
+    ov::Tensor image = pipe.generate(prompt,
+        ov::genai::generator(std::make_shared<ov::genai::CppStdGenerator>(42)),
+        ov::genai::width(512),
+        ov::genai::height(896),
+        ov::genai::num_inference_steps(20));
```

Comment:
why do we need to change default number of inference steps? E.g. for LCM it's 4 and here you increase it 4x times

the same question for initial Lora sample

Reply:
@ilya-lavrenov, it wasn't changed for LoRA specifically, setting number of iterations was here starting with the original text2image sample without LoRA, and it is common for all text2image samples now. It was just borrowed from the original sample to LoRA samples. If we really can omit it, it would simplify the sample, but do it for all text2image samples in this case. Why was it set initially? For illustrative purposes?

Reply:
yes, I forgot that all samples already contain this parameter.

Initially it was added to limit number of iterations in SD SDXL cases where 50 iterations are set by default. I think we can separately limit a number of iterations up to 20, but keep lower values as it (e.g. 4 for LCM case)